### PR TITLE
Report conflicting projects in updateable list

### PR DESCRIPTION
**Pull request by ehelms**
Example output:

Before:
```
Found conflict for rubygem-concurrent-ruby: 1.1.10 != 1.2.2
Found conflict for rubygem-tilt: 2.2.0 != 2.1.0
Found conflict for rubygem-fast_gettext: 1.8.0 != 2.3.0
Found conflict for rubygem-unicode-display_width: 1.8.0 != 2.4.2
Found conflict for rubygem-version_gem: 1.1.3 != 1.1.2
```

After:
```
Found conflict for rubygem-concurrent-ruby: 1.1.10 (foreman) != 1.2.2 (smart-proxy)
Found conflict for rubygem-tilt: 2.2.0 (foreman) != 2.1.0 (smart-proxy)
Found conflict for rubygem-fast_gettext: 1.8.0 (foreman) != 2.3.0 (hammer-cli)
Found conflict for rubygem-unicode-display_width: 1.8.0 (smart-proxy) != 2.4.2 (hammer-cli)
Found conflict for rubygem-version_gem: 1.1.3 (foreman) != 1.1.2 (hammer-cli)
```

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl I'd appreciate your thoughts on this updated output.

File: list_updatable_packages [hunk 1]
```diff
@@ -103,15 +103,15 @@ def find_gemfile_lock_specs(ci_job: str, subdir: str) -> Generator[Tuple[Spec, s
         if path.is_file():
             spec = Spec(path)
             if spec.is_updateable:
-                yield spec, gem['version']
+                yield spec, gem['version'], project
 
 
 def find_packaged_specs() -> Generator[Tuple[Spec, None], None, None]:
```
Comment (ekohl) [inline list_updatable_packages:109]:
```suggestion
def find_packaged_specs() -> Generator[Tuple[Spec, None, None], None, None]:
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Looks good, just need to fix the type annotations so `mypy` is happy.

File: list_updatable_packages (hunk 1 above)
Comment (ehelms) [reply list_updatable_packages:109]: Curious, I am surprise issues like this did not throw errors.

Comment (ekohl) [reply list_updatable_packages]: Type definitions aren't enforced at runtime, but can be used by external tools (and introspected at runtime). So your IDE can provide better auto completion suggestions. https://pydantic.dev/ is a library that does use the type definitions at runtime to (de)serialize data, but also validate it.

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**